src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy

```diff
@@ -80,13 +81,32 @@ class GatlingPlugin implements Plugin<Project> {
                 gatling "io.gatling.highcharts:gatling-charts-highcharts:${p.extensions.getByType(GatlingPluginExtension).toolVersion}"
 
                 if (gatlingExt.includeMainOutput) {
-                    gatlingImplementation project.sourceSets.main.output
+                    gatlingImplementation p.sourceSets.main.output
                 }
                 if (gatlingExt.includeTestOutput) {
-                    gatlingImplementation project.sourceSets.test.output
+                    gatlingImplementation p.sourceSets.test.output
                 }
             }
+
+            if (p.repositories.isEmpty() || shouldAddMavenCentral(p)) {
+                p.repositories {
+                    mavenCentral(name: "gatlingMavenCentral")
+                }
+            }
+        }
+    }
+
+    private static boolean shouldAddMavenCentral(Project project) {
+        def resolutionRes = project.configurations.gatling.copyRecursive().incoming.resolutionResult
+        def deps = resolutionRes.allDependencies
+
+        if (deps.size() > 1 && deps.any { it instanceof UnresolvedDependencyResult }) {
+            return true
+        } else if (deps.size() == 1 && project.repositories.every { it.url.toString() != "https://repo.maven.apache.org/maven2/" }) {
+            return true
         }
+
+        return false
```

Comment:
Sorry, I'm not sure about the reasoning for those different branches.

It think this would deserve either some comments or having intermediate booleans with explicit names and combine them into one single boolean operation instead of returning true/false.

Could you please elaborate?

Reply:
I'd prefer to add comments to code, at least for now. Don't you mind ?

Reply:
Fine with me.